### PR TITLE
feat: Add possibility to select all files in offline root

### PR DESCRIPTION
**Pull request by BaptGrv**

### CODE REVIEW

**Review by adrien-coye — COMMENTED** [review]
I just did a review of the code. LGTM.
I did found some issues regarding the actions available when using the "select all" compared to manual select all the element. I'll show you.

File: kDrive/UI/Controller/Menu/OfflineFilesViewModel.swift [hunk 1]
```diff
@@ -53,4 +54,20 @@ class OfflineFilesViewModel: FileListViewModel {
...
+    override func barButtonPressed(sender: Any?, type: FileListBarButtonType) {
+        let viewModel = multipleSelectionViewModel
+        viewModel?.isSelectAllModeEnabled = true
+        viewModel?.rightBarButtons = [.loading]
+        viewModel?.onSelectAll?()
+        if type == .selectAll {
+            let files = driveFileManager.database.fetchResults(ofType: File.self) { lazyCollection in
+                lazyCollection.filter("isAvailableOffline = true")
+            }
+            viewModel?.selectedCount = files.count
```
Comment (adrien-coye) [inline kDrive/UI/Controller/Menu/OfflineFilesViewModel.swift:67]: You need to track actual selected files in the `multipleSelectionViewModel` with `selectedItems` for the actions to work.
You can create a dedicated function `didSelectFiles` in `MultipleSelectionFileListViewModel` to set the selection and count.

**Review by adrien-coye — CHANGES_REQUESTED** [review]
Here is how you can fix the actions (see comment)

**Review by adrien-coye — COMMENTED** [review]
Good job 👍 
I did test all actions, all good.
~~We have one corner case to discuss in iOS meeting before I can merge the PR.~~
Once we disable the download when not availlable, we can merge this.

**Review by adrien-coye — APPROVED** [review]
Tested all actions. LGTM 👍
